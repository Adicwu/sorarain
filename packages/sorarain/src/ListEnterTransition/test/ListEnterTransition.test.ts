import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import { h } from 'vue'
import ListEnterTransition from '../ListEnterTransition.vue'

describe('ListEnterTransition.vue', () => {
  test('visible事件 正确推送', () => {
    const wrapper = mount(ListEnterTransition)

    expect(wrapper.emitted('visible')).toBeUndefined()
  })
  test('默认插槽运行正常', () => {
    const wrapper = mount(ListEnterTransition, {
      slots: {
        default: () =>
          [1, 2, 3].map((item) =>
            h(
              'div',
              {
                key: item
              },
              item
            )
          )
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
