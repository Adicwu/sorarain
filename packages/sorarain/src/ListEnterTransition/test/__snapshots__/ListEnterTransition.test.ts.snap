// Vitest Snapshot v1

exports[`ListEnterTransition.vue > 默认插槽运行正常 1`] = `
"<transition-group-stub>
  <div>1</div>
  <div>2</div>
  <div>3</div>
</transition-group-stub>"
`;

exports[`should render default slot correctly 1`] = `
"<transition-group-stub>
  <div>1</div>
  <div>2</div>
  <div>3</div>
</transition-group-stub>"
`;

exports[`should render icon slot correctly 1`] = `
"<transition-group-stub>
  <div>1</div>
  <div>2</div>
  <div>3</div>
</transition-group-stub>"
`;

exports[`默认插槽运行正常 1`] = `
"<transition-group-stub>
  <div>1</div>
  <div>2</div>
  <div>3</div>
</transition-group-stub>"
`;
